packages/web3-validator/src/keywords/eth.ts

```diff
@@ -0,0 +1,116 @@
+import { AnySchemaObject, FuncKeywordDefinition, SchemaCxt } from 'ajv';
+import { Web3ValidatorError } from '../errors';
+import {
+	DataValidateFunction,
+	DataValidationCxt,
+	ValidInputTypes,
+	Web3ValidationErrorObject,
+} from '../types';
+import {
+	isBoolean,
+	isString,
+	isBytes,
+	isInt,
+	isUInt,
+	isAddress,
+	isValidEthType,
+	isBloom,
+} from '../validation';
+
+const createErrorObject = (
+	message: string,
+	value: unknown,
+): Partial<Web3ValidationErrorObject> => ({
+	message,
+	keyword: 'eth',
+	params: { value },
+});
+
+export const metaSchema = {
+	title: 'Web3 Ethereum Compatible Types',
+	type: 'string',
+};
+
+const compile = (
+	type: string,
+	parentSchema: AnySchemaObject,
+	it: SchemaCxt,
+): DataValidateFunction => {
+	const typePropertyPresent = Object.keys(parentSchema).includes('type');
+
+	if (typePropertyPresent) {
+		throw new Web3ValidatorError([
+			{
+				keyword: 'eth',
+				message: 'Either "eth" or "type" can be presented in schema',
+				params: { eth: type },
+				instancePath: '',
+				schemaPath: it.schemaPath.str ?? '',
+			},
+		]);
+	}
+
+	if (!isValidEthType(type)) {
+		throw new Web3ValidatorError([
+			{
+				keyword: 'eth',
+				message: `Eth data type "${type}" is not valid`,
+				params: { eth: type },
+				instancePath: '',
+				schemaPath: it.schemaPath.str ?? '',
+			},
+		]);
+	}
+
+	const validate: DataValidateFunction = (
+		data: ValidInputTypes,
+		// TODO: Could be useful in future
+		// eslint-disable-next-line @typescript-eslint/no-unused-vars
+		_dataCxt?: DataValidationCxt,
```

Comment:
We should replace this with a comment that mentions `_dataCxt` is an option and when it might be useful, but we should not have unused code that may be used in the future